mountpoint-s3-crt/src/s3/endpoint_resolver.rs

```diff
@@ -307,4 +322,17 @@ mod test {
             ResolverError::EndpointNotResolved("A region must be set when sending requests to S3.".to_owned())
         );
     }
+
+    #[test_case("arn:aws:s3-object-lambda:eu-west-1:AccountID:accesspoint/olap-bucket-test", "{\"authSchemes\":[{\"disableDoubleEncoding\":true,\"name\":\"sigv4\",\"signingName\":\"s3-object-lambda\",\"signingRegion\":\"eu-west-1\"}]}"; "OLAP ARN")]
+    #[test_case("arn:aws:s3:us-east-2:AccountID:accesspoint/access-point-test", "{\"authSchemes\":[{\"disableDoubleEncoding\":true,\"name\":\"sigv4\",\"signingName\":\"s3\",\"signingRegion\":\"us-east-2\"}]}"; "Access Point ARN")]
+    #[test_case("arn:aws:s3::accountID:accesspoint/s3-bucket-test.mrap", "{\"authSchemes\":[{\"disableDoubleEncoding\":true,\"name\":\"sigv4a\",\"signingName\":\"s3\",\"signingRegionSet\":[\"*\"]}]}"; "MRAP")]
+    fn test_endpoint_properties_for_arn(arn: &str, endpoint_property: &str) {
+        let new_allocator = Allocator::default();
+        let endpoint_rule_engine = RuleEngine::new(&new_allocator).unwrap();
+        let mut endpoint_request_context = RequestContext::new(&new_allocator).unwrap();
+        test_endpoint_resolver_init(arn, "us-east-1", &new_allocator, &mut endpoint_request_context);
+        let resolved_endpoint = endpoint_rule_engine.resolve(endpoint_request_context).unwrap();
+        let property = resolved_endpoint.get_properties();
+        assert_eq!(property, endpoint_property);
```

Comment:
This test should also call `get_auth_scheme_for_endpoint` (or whatever you rename it to) and check the results. Might be easier without the `test_case` macro, not sure though.

Reply:
This is inside CRT right? I can add tests in `mountpoint-client` for the `AuthScheme`.